### PR TITLE
Adding api_key and model_name parameters to the restful benchmark

### PR DESCRIPTION
**Pull request by NiuBlibing**
## Motivation

Adding api_key and model_name parameters to the restful benchmark.

### CODE REVIEW

File: benchmark/profile_restful_api.py [hunk 1]
```diff
@@ -66,21 +66,28 @@ def __init__(self,
...
                  **kwargs):
         self.tokenizer = Tokenizer(tokenzier_path)
         self.server_addr = server_addr
         self.temperature = temperature
         self.top_p = top_p
         self.csv = csv
-        client = APIClient(self.server_addr)
-        self.model_name = client.available_models[0]
+        self.api_key = api_key
+        client = APIClient(self.server_addr, api_key=self.api_key)
```
Comment (zhyncs) [inline benchmark/profile_restful_api.py:78]: What's the purpose of adding the `api_key` in the benchmark script, and in what scenarios would it be used?

Comment (NiuBlibing) [reply benchmark/profile_restful_api.py:78]: 1. To test the service speed from provider such as openai, some cloud.
2. Users often have only one set of environmen due to lack of resources, which needs a api_key to access.

Comment (zhyncs) [reply benchmark/profile_restful_api.py:78]: > 1. To test the service speed from provider such as openai, some cloud.
> 2. Users often have only one set of environmen due to lack of resources, which needs a api_key to access.

make sense

File: benchmark/profile_restful_api.py [hunk 2]
```diff
@@ -66,21 +66,28 @@ def __init__(self,
...
+        if model_name is None:
+            self.model_name = client.available_models[0]
+            print(f"using model:{self.model_name}")
```
Comment (zhyncs) [inline benchmark/profile_restful_api.py:81]: A small change, it is best for the print format to be consistent with the existing form in the script.

```python3
print(f'using model: {self.model_name}\n')
```

Comment (NiuBlibing) [reply benchmark/profile_restful_api.py]: Ok.

**Review by zhyncs — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**